OpenKeychain/src/main/java/org/sufficientlysecure/keychain/securitytoken/SecurityTokenHelper.java

```diff
@@ -704,35 +713,36 @@ private String tryPin(int mode, byte[] pin) throws IOException {
      * Afterwards, the token is reactivated.
      */
     public void resetAndWipeToken() throws IOException {
-        String accepted = "9000";
-
         // try wrong PIN 4 times until counter goes to C0
         byte[] pin = "XXXXXX".getBytes();
         for (int i = 0; i <= 4; i++) {
-            String response = tryPin(0x81, pin);
-            if (response.equals(accepted)) { // Should NOT accept!
-                throw new CardException("Should never happen, XXXXXX has been accepted!", parseCardStatus(response));
+            ResponseAPDU response = tryPin(0x81, pin);
+            if (response.getSW() == APDU_SW_SUCCESS) { // Should NOT accept!
+                throw new CardException("Should never happen, XXXXXX has been accepted!", response.getSW());
             }
         }
 
         // try wrong Admin PIN 4 times until counter goes to C0
         byte[] adminPin = "XXXXXXXX".getBytes();
         for (int i = 0; i <= 4; i++) {
-            String response = tryPin(0x83, adminPin);
-            if (response.equals(accepted)) { // Should NOT accept!
-                throw new CardException("Should never happen, XXXXXXXX has been accepted", parseCardStatus(response));
+            ResponseAPDU response = tryPin(0x83, adminPin);
+            if (response.getSW() == APDU_SW_SUCCESS) { // Should NOT accept!
+                throw new CardException("Should never happen, XXXXXXXX has been accepted", response.getSW());
             }
         }
 
         // reactivate token!
-        String reactivate1 = "00" + "e6" + "00" + "00";
-        String reactivate2 = "00" + "44" + "00" + "00";
-        String response1 = communicate(reactivate1);
-        String response2 = communicate(reactivate2);
-        if (!response1.equals(accepted) || !response2.equals(accepted)) {
-            throw new CardException("Reactivating failed!", parseCardStatus(response1));
-        }
 
+        CommandAPDU reactivate1 = new CommandAPDU(0x00, 0xE6, 0x00, 0x00);
```

Comment:
while looking at this I remember something (not related to your refactorings):
Can you please add a comment here:
``// NOTE: keep the order here! First execute _both_ reactivate commands. Before checking _both_ response
// If a token is in a bad state and reactivate1 fails, it could still be reactivated with reactivate2``